### PR TITLE
Add generation and verification of schnorr signature

### PR DESCRIPTION
**Pull request by rantan**
## Specification

The spec document is here:
* https://github.com/chaintope/tapyrus-core/blob/master/doc/tapyrus/schnorr_signature.md

and test vectors are created by:
* https://github.com/chaintope/tapyrus-schnorr-signature-test-vectors

### CODE REVIEW

File: src/util/prime.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+use rug::integer::Order;
+
+/// Prime number for secp256k1 field element.
+pub const P: [u8; 32] = [
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xfe, 0xff, 0xff, 0xfc, 0x2f
+];
+
```
Comment (azuchi) [inline src/util/prime.rs:19]: This field element should be managed by rust-secp256k1?

Comment (rantan) [reply src/util/prime.rs:19]: Agree, [this file](https://github.com/rust-bitcoin/rust-secp256k1/blob/master/src/constants.rs) can be hold this value. 
I'm going to try to create PR.

File: src/util/mod.rs [hunk 2]
```diff
@@ -34,6 +34,9 @@ pub mod merkleblock;
 pub mod misc;
 pub mod psbt;
 pub mod uint;
+pub mod signature;
+pub mod prime;
+pub mod rfc7969;
```
Comment (Yamaguchi) [inline src/util/mod.rs:39]: -> rfc6979

File: src/util/rfc7969.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    let mut rng = RFC6879::new(&keydata[..]);
+    let mut ret: [u8; 32] = [0u8; 32];
+    for _ in 0..=counter {
+        ret = rng.generate();
+    }
+
+    ret
+}
+
+struct RFC6879 {
```
Comment (Yamaguchi) [inline src/util/rfc7969.rs:38]: -> RFC6979

File: src/util/rfc7969.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2020 Chaintope Inc.
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+//! RFC6879
```
Comment (Yamaguchi) [inline src/util/rfc7969.rs:5]: -> RFC6979

File: src/util/prime.rs (hunk 1 above)
Comment (azuchi) [reply src/util/prime.rs:19]: OK. For now, merge this, but replace when the pull request is merged.

**Pull request merged**